htdocs/product/price.php

```diff
@@ -292,7 +292,7 @@
 			$newlocaltax2_type = GETPOST('localtax2_type', 'array');
 
 			//Shall we generate prices using price rules?
-			$object->price_autogen = GETPOST('usePriceRules') == 'on';
+			$object->price_autogen = (int) GETPOST('usePriceRules') == 'on';
```

Comment:
Should this not be (int) (GETPOST() == 'on') ?

Reply:
@lamrani002 i think @mdeweerd is right...